plugins/pi4j-plugin-linuxfs/src/main/java/com/pi4j/plugin/linuxfs/provider/i2c/LinuxFsI2CProviderImpl.java

```diff
@@ -56,6 +56,8 @@ public synchronized I2C create(I2CConfig config) {
         LinuxFsI2CBus i2CBus = this.i2CBusMap.computeIfAbsent(config.getBus(), busNr -> new LinuxFsI2CBus(config));
         // create new I/O instance based on I/O config
         LinuxFsI2C i2C = new LinuxFsI2C(i2CBus, this, config);
+        // Workaround, needed if first LinuxFsI2C usage is ioctl (readRegister or writeRegister)
+        i2C.read();
```

Comment:
Is there not an init of a buffer in this method, which then makes the ioctl work?

Reply:
For instance i can see this:

        this.lastAddress = i2c.device();
        this.file.ioctl(I2CConstants.I2C_SLAVE, i2c.device() & 0xFF);

so the very first thing that happens, is that we set the lastAddress, then immediately perform an ioctl... So maybe this is missing?

Reply:
ok, i do see that the readRegister() does this too...